src/ucp/proto/proto_am.inl

```diff
@@ -101,6 +132,10 @@ ucs_status_t ucp_do_am_bcopy_multi(uct_pending_req_t *self, uint8_t am_id_first,
             if ((packed_len == UCS_ERR_NO_RESOURCE) &&
                 (req->send.lane != req->send.pending_lane)) {
                 /* switch to new pending lane */
+                status = ucp_am_copy_user_header_to_internal_buf(req);
```

Comment:
Can you do this in UCP AM specific functions instead? (which invoke these generic functions)
Then you could remove one of the branches from `ucp_am_copy_user_header_to_internal_buf`

Reply:
Yes I think so. It's will require changes in more functions, (ucp_am_short_handle_status_from_pending)...
I will post an update soon.